spec/rubocop/cop/layout/space_before_brackets_spec.rb

```diff
@@ -4,26 +4,52 @@
   subject(:cop) { described_class.new(config) }
 
   context 'when referencing' do
-    it 'registers an offense and corrects when using space between receiver and left brackets' do
+    it 'registers an offense and corrects when using space between lvar receiver and left brackets' do
       expect_offense(<<~RUBY)
+        collection = do_something
         collection [index_or_key]
                   ^ Remove the space before the opening brackets.
       RUBY
 
       expect_correction(<<~RUBY)
+        collection = do_something
         collection[index_or_key]
       RUBY
     end
 
-    it 'does not register an offense when not using space between receiver and left brackets' do
+    it 'registers an offense and corrects when using space between ivar receiver and left brackets' do
```

Comment:
Seems to me this should cover `gvars` and `cvars` as well.

Reply:
I added these tests.